documentation/openapi_backend_routes.yaml

```diff
@@ -0,0 +1,1148 @@
+    post:
+      tags:
+      - code
+      summary: runs the source code and stdin for edugator
+      operationId: postRun
+      requestBody:
+        description: Run object that needs to be added to run queue
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/Run'
+        required: true
+      responses:
+        201:
+          description: Successful operation
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/JudgeToken'
+        400:
+          description: Invalid input
+          content: {}
+        422:
+          description: Language Id cannot be blank
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Judge0LanguageError'
+        503:
+          description: queue is full error
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Judge0QueueError'
+      x-codegen-request-body-name: body
+  /code/run/{runId}:
+    get:
+      tags:
+      - code
+      summary: gets the output of the source code run from token
+      operationId: getRun
+      parameters:
+      - name: runId
+        in: path
+        description: ID of run to return
+        required: true
+        schema:
+          type: string
+      responses:
+        200:
+          description: successful operation
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/SubmissionStats'
+  /course/{courseId}/problem:
+    get:
+      tags:
+      - problem
+      summary: Gets all the problems for COP3530 course
+      operationId: getProblem
+      parameters:
+      - name: courseId
+        in: path
+        description: ID of course
+        required: true
+        schema:
+          type: string
+      responses:
+        200:
+          description: successful operation
+          content:
+            application/json:
+              schema:
+                type: array
+                items:
+                  $ref: '#/components/schemas/Problem'
+  /course/{courseId}/module/{moduleId}/problem/:
+    get:
+      tags:
+      - problem
+      summary: Gets all the non hidden problems for COP3530 course given module id
+      operationId: getStudentProblemsByModuleId
+      parameters:
+      - name: moduleId
+        in: path
+        description: ID of module to return
+        required: true
+        schema:
+          type: string
+      - name: courseId
+        in: path
+        description: Id of course
+        required: true
+        schema:
+          type: string
+      responses:
+        200:
+          description: successful operation
+          content:
+            application/json:
+              schema:
+                type: array
+                items:
+                  $ref: '#/components/schemas/Problem'
+  /problem:
```

Comment:
should we rename the problem routes to something else - as we have two types now: lessons vs coding problems. Or there should be some means to distinguish the two